actors/builtin/miner/miner_test.go

```diff
@@ -4640,6 +4653,70 @@ func (h *actorHarness) preCommitSector(rt *mock.Runtime, params *miner.PreCommit
 	return h.getPreCommit(rt, params.SectorNumber)
 }
 
+type preCommitBatchConf struct {
+	sectorWeights []market.SectorWeights
+}
+
+func (h *actorHarness) preCommitSectorBatch(rt *mock.Runtime, params *miner.PreCommitSectorBatchParams, conf preCommitBatchConf, first bool) []*miner.SectorPreCommitOnChainInfo {
+	rt.SetCaller(h.worker, builtin.AccountActorCodeID)
+	rt.ExpectValidateCallerAddr(append(h.controlAddrs, h.owner, h.worker)...)
+	{
+		expectQueryNetworkInfo(rt, h)
+	}
+	sectorDeals := make([]market.SectorDeals, len(params.Sectors))
+	sectorWeights := make([]market.SectorWeights, len(params.Sectors))
+	anyDeals := false
+	for i, sector := range params.Sectors {
+		sectorDeals[i] = market.SectorDeals{
+			SectorExpiry: sector.Expiration,
+			DealIDs:      sector.DealIDs,
+		}
+
+		if len(conf.sectorWeights) > i {
+			sectorWeights[i] = conf.sectorWeights[i]
+		} else {
+			sectorWeights[i] = market.SectorWeights{
+				DealSpace:          0,
+				DealWeight:         big.Zero(),
+				VerifiedDealWeight: big.Zero(),
+			}
+		}
+		// Sanity check on expectations
+		sectorHasDeals := len(sector.DealIDs) > 0
+		dealTotalWeight := big.Add(sectorWeights[i].DealWeight, sectorWeights[i].VerifiedDealWeight)
+		require.True(h.t, sectorHasDeals == !dealTotalWeight.Equals(big.Zero()), "sector deals inconsistent with configured weight")
+		require.True(h.t, sectorHasDeals == (sectorWeights[i].DealSpace != 0), "sector deals inconsistent with configured space")
+		anyDeals = anyDeals || sectorHasDeals
+	}
+	if anyDeals {
+		vdParams := market.VerifyDealsForActivationParams{
+			Sectors: sectorDeals,
+		}
+		vdReturn := market.VerifyDealsForActivationReturn{
+			Sectors: sectorWeights,
+		}
+		rt.ExpectSend(builtin.StorageMarketActorAddr, builtin.MethodsMarket.VerifyDealsForActivation, &vdParams, big.Zero(), &vdReturn, exitcode.Ok)
+	}
+	st := getState(rt)
+	if st.FeeDebt.GreaterThan(big.Zero()) {
+		rt.ExpectSend(builtin.BurntFundsActorAddr, builtin.MethodSend, nil, st.FeeDebt, nil, exitcode.Ok)
+	}
+
+	if first {
```

Comment:
suggestions: if it's not too hard to do for this new method and it makes sense to you, you could add this param to the config.  I think that might be the place it belongs.

Reply:
Easy, done.